test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeSystemArraycopyTests.java

```diff
@@ -0,0 +1,785 @@
+			dst[i] = new SomeValueClass(i*25);
+		}
+		testOBJOBJ(ifVtArraySrc, dst);
+		checkResults(ifVtArraySrc, dst);
+
+		for (int i=0; i < ARRAY_SIZE; i++) {
+			dst[i] = new SomePrimitiveValueClass(i*26);
+		}
+		testOBJOBJ(ifPrimitiveVtArraySrc, dst);
+		checkResults(ifPrimitiveVtArraySrc, dst);
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy22() throws Throwable {
+
+		try {
+			initArraysForNPETest(); // ifArray3[ARRAY_SIZE/2] is NULL
+			testIFPVT(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.NullPointerException npe1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForNPETest();
+				testIFPVT(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.NullPointerException npe2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
+			}
+		}
+
+		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy23() throws Throwable {
+
+		try {
+			initArraysForNPETest(); // ifArray3[ARRAY_SIZE/2] is NULL
+			testIFIF(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.NullPointerException npe1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForNPETest();
+				testIFIF(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.NullPointerException npe2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
+			}
+		}
+
+		Assert.fail("Expect a NullPointerException. No exception or wrong kind of exception thrown");
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy24() throws Throwable {
+
+		try {
+			initArraysForArrayStoreChkExceptionTest();  // ifArray3[ARRAY_SIZE/2] is SomeIdentityClass
+			testIFPVT(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.ArrayStoreException ase1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForArrayStoreChkExceptionTest();
+				testIFPVT(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.ArrayStoreException ase2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+				// pass
+			return;
+			}
+		}
+
+		Assert.fail("Expect a ArrayStoreException. No exception or wrong kind of exception thrown");
+	}
+
+	@Test(priority=1)
+	static public void testSystemArrayCopy25() throws Throwable {
+
+		try {
+			initArraysForArrayStoreChkExceptionTest();  // ifArray3[ARRAY_SIZE/2] is SomeIdentityClass
+			testIFIF(ifArray3, primitiveVtArrayDst);
+		} catch (java.lang.ArrayStoreException ase1) {
+			try {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
+
+				initArraysForArrayStoreChkExceptionTest();
+				testIFIF(ifArray3, primitiveVtArrayDst);
+			} catch (java.lang.ArrayStoreException ase2) {
+				checkResultsPartial(ifArray3, primitiveVtArrayDst, ARRAY_SIZE/2);
```

Comment:
`checkResultsPartial` is only testing the contents of the destination array up to the point where the `ArrayStoreException` or `NullPointerException` is expected to occur.  You'll also need to test the contents from that point to the last element are unchanged from before the call to `System.arraycopy` to verify that the copying actually stopped.